File: exercicios/lista01/q4-cinema.cpp
#include <iostream>
using namespace std;

class Ingresso {
    public:
    string dia;
    short hora;

    Ingresso(string dia_arg, short hora_arg) {
        dia = dia_arg;
        hora = hora_arg;
    }

    short calcula_ingresso() {
        if (dia == "seg" || dia == "ter" || dia == "qui") {
            short valor = 16;
            if (hora == 0 || hora >= 17) {
                valor = valor * 1.5;
            }

            return valor;
        } 
        else if (dia == "qua") {
            short valor = 8;
            return valor;
        }
        else if (dia == "sex" || dia == "sab" || dia == "dom") {
            short valor = 20;
            if (hora == 0 || hora >= 17) {
                valor = valor * 1.5;
            }
            return valor;
        }
    }
};

class UI {
    public:
    string cabecalho;

    UI() {
        cabecalho = "========= Bem-vindo(a) ao seletor de ingressos =========";
    }

    void main() {
        cout << cabecalho << endl;
        int op = 0;
        while (op != 2) {
            op = this->menu();
            if (op == 1) {
                this->ingresso();
            };
        };
        cout << "========= Até a próxima =========" << endl;
    }

    int menu() {
        int op;
        short ARRAY_LENGTH = 2;
        string opcoes[ARRAY_LENGTH] = {"1 - Ingressos", "2 - Sair"};

        for (int i=0; i < ARRAY_LENGTH; ++i) {
            cout << opcoes[i] << endl;
        };

        cout << "Escolha um valor:" << endl;
        cin >> op;
        return op;
    }

    void ingresso() {
        cout << "Ingresso" << endl;
    }
};

int main() {
    UI ui;
    ui.main();
    return 0;
}

File: exercicios/lista01/q4-cinema.py
class Ingresso:
    dia = ""
    horario = None
    def __init__(self, dia, horario):
        self.dia = dia
        self.horario = horario
    
    def inteira(self):
        if self.dia in ["seg", "ter", "qui"]:
            valor = 16
            if self.horario == 0 or self.horario >= 17:
                return valor * 1.5
            return valor
        
        elif self.dia == "qua":
            return 8
        
        elif self.dia in ["sex", "sab", "dom"]:
            valor = 20
            if self.horario == 0 or self.horario >= 17:
                return valor * 1.5
            return valor
    
    def meia(self):
        if self.dia == "qua":
            return 8
        return self.inteira()/2


class UI:
    def main():
        op = 0
        while op != 2:
            op = UI.menu()
            if op == 1:
                UI.ingresso()
        print("Saindo...")
        
    def menu():
        opcoes = ["1 - Ingressos", "2 - Sair"]

        for i in opcoes:
            print(i)

        op = int(input("Escolha uma opção: "))
        return op
    
    def ingresso():
        dia = input("Informe o dia: ")
        horario = int(input("Informe o horário: "))
        ingr = Ingresso(dia, horario) 
        preco_inteira = ingr.inteira()
        preco_meia = ingr.meia()
        print(f"Preço da inteira: R${preco_inteira:0.2f}")
        print(f"Preço da meia: R${preco_meia:0.2f}")

UI.main()